bool activate
---
int32 distance